### PR TITLE
19 sidebar component

### PR DESCRIPTION
**Pull request by Fatmasiam**
#19

### CODE REVIEW

File: Client/src/components/common/Link/index.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.sublink{
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.css]: again class name, always be more specific. 
```
sidebar-sublink
```

File: Client/src/components/common/Link/index.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  font-size: 16px;
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.css:2]: font-size by default 16px

File: Client/src/components/common/Sidebar/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import './index.css'
+import Linklst from '../LinkList';
+
+
+export default function Sidebar({ linklist }) {
+  return (
+    <div className='sidebar'>
+      <img src='https://i.imgur.com/m6QdXqO.png' alt='AMDIEAST logo' className='sidebar-desktop-logo' width='200px' />
+      <h3 className='sidebar-title' >AMIDEAST EL</h3>
+      <hr className='hr' />
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js]: again generic class name

File: Client/src/components/common/Sidebar/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      <br />
+      <ul className='sidebar-linklist'>
+        {linklist.map((linklist, index) => <Linklst key={index} links={linklist} />)}
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js]:
```suggestion
        {linklist.map((linklist, index) => <Linklist key={index} links={linklist} />)}
```

File: Client/src/components/common/Sidebar/index.css [hunk 5]
```diff
@@ -0,0 +1,25 @@
+.sidebar {
+  width: 20%;
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.css:2]: good idea

File: Client/src/components/common/Sidebar/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import './index.css'
+import Linklst from '../LinkList';
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js]:
```suggestion
import Linklist from '../LinkList';
```

File: Client/src/components/common/LinkList/index.css [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+.sidebar-ul-li-ul {
+  list-style-type: none;
+  margin-top: 6px;
+}
+
+.sidebar-icon {
+  padding-right: 7px;
+}
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.css]: new line

File: Client/src/components/common/Link/index.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import './index.css'
+
+export default function Link({ link }) {
+  return (
+    <li className='sublink'> {link.title} </li>
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.js]:
```suggestion
<Link  to={link.to}> 
      <li className='sublink'> {link.title} </li>
</Link>
  ```

File: Client/src/components/common/LinkList/index.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Link from '../Link';
+import './index.css';
+
+export default function Linklist({ links }) {
+  return (
+    <li className='sidebar-li-links'>
+      <i className={links.fontawesomeClass} ></i>
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.js]:
```suggestion
      <i className={links.fontAwesome} ></i>
```

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: Client/src/components/common/Link/index.css (hunk 2 above)
Comment (Fatmasiam) [reply Client/src/components/common/Link/index.css:2]: @knanahassouna1  the default is ```18px```

File: Client/src/components/common/Sidebar/index.js (hunk 3 above)
Comment (Fatmasiam) [reply Client/src/components/common/Sidebar/index.js]: @knanahassouna1  do you mean to hr?

File: Client/src/components/common/Link/index.css (hunk 2 above)
Comment (kannanhassouna) [reply Client/src/components/common/Link/index.css:2]: https://www.google.com/search?q=default+font+size+of+html&oq=default+font-size+&aqs=chrome.1.69i57j0l5.5205j0j1&sourceid=chrome&ie=UTF-8

File: Client/src/components/common/LinkList/index.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import Link from '../Link';
+import './index.css';
+
+export default function Linklist({ links }) {
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.js]:
```suggestion
export default function Linklist({ links: { links}  }) {
```

File: Client/src/components/common/LinkList/index.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <li className='sidebar-li-links'>
+      <i className={links.fontawesome} ></i>
+      {links.title}
+      <ul className='sidebar-ul-li-ul'>
+        {links.links.map((link, index) => {
+          console.log(link.title, link.to)
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.js]: remove console.log

File: Client/src/components/common/LinkList/index.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import Link from '../Link';
+import './index.css';
+
+export default function Linklist({ links }) {
+  return (
+    <li className='sidebar-li-links'>
+      <i className={links.fontawesome} ></i>
+      {links.title}
+      <ul className='sidebar-ul-li-ul'>
+        {links.links.map((link, index) => {
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.js]:
```suggestion
        {links.links.map(({ title, to }, index) => {
```

**Review by kannanhassouna — APPROVED** [review]

File: Client/src/components/common/Link/index.js [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './index.css'
+import { Link } from 'react-router-dom';
+
+export default function link({ title, to }) {
+  console.log(title)
+  return (to ?
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Link/index.js:7]: remove the console.log(title)
line 6

File: Client/src/components/common/Link/index.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './index.css'
+import { Link } from 'react-router-dom';
+
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Link/index.js:3]: ### For the import arrangement
- first: import third-party modules
- second: import your files 
and between them add a new line

Comment (AbdallahAmmar96) [inline Client/src/components/common/Link/index.js:3]:
```suggestion
import React from 'react';
import { Link } from 'react-router-dom';

import './index.css';
```

Comment (AbdallahAmmar96) [reply Client/src/components/common/Link/index.js:3]: see code below

File: Client/src/components/common/LinkList/index.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+          console.log(link.title, link.to)
+          return < Link key={index} title={link.title} to={link.to} />
+        }
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/LinkList/index.js:14]:
```suggestion
        return <Link key={index} title={link.title} to={link.to} />
        }
```
indentation

File: Client/src/components/common/Sidebar/index.js [hunk 16]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import './index.css'
+import Linklist from '../LinkList';
+
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js:2]: same comment above for the arrangement

File: Client/src/components/common/Sidebar/index.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+      </ul>
+    </div>
+  )
+}
+
+Sidebar.defaultProps = {
+  linklist: [
+    {
+      fontawesome: 'fas fa-home sidebar-icon',
+      title: 'Home',
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js]: Put the defaultProps in an external static file and call it here

**Review by AbdallahAmmar96 — APPROVED** [review]

**Pull request merged**